### PR TITLE
docs(aio): update content after migration fixes

### LINKED ISSUE #15592: aio: ensure overview docs can be viewed
See https://github.com/angular/angular/pull/15570#discussion_r108780930

### PR DESCRIPTION
**Pull request by petebacondarwin**
Fixes #15592

### CODE REVIEW

**Comment by petebacondarwin** [comment]
This PR fixes up most (if not all) of the crazy non-rendered markdown appearing in the docs.

@IgorMinar or @gkalpak PTAL

**Comment by IgorMinar** [comment]
I guess this looks good ☺ 

I scrolled through and didn't see anything super odd.

**Review by IgorMinar — APPROVED** [review]

File: aio/content/glossary.md [hunk 1]
```diff
@@ -154,6 +165,48 @@ operations and supporting declaration syntax.
...
 
+~~~
+
+
+
+{@a decorator}
+
+
+{@a decoration}
+## Decorator | decoration
```
Comment (gkalpak) [inline aio/content/glossary.md]: This section is rendered in a way that seems wrong (https://pr15635-ce8a2b4fc829ca18647dc40f2543f8a6a7e5d6eb.ngbuilds.io/guide/glossary#decoration).
(E.g. code block is formatted as inline code, there is an empty alert, etc.)

File: aio/content/glossary.md [hunk 2]
```diff
@@ -421,7 +473,6 @@ Read more in the [Lifecycle Hooks](!{docsLatest}/guide/lifecycle-hooks) page.
 
 
 
-
```
Comment (gkalpak) [inline aio/content/glossary.md]: I am sure it is not related to this removed line, but the "Module" section is also formatted incorrectly.

**Review by gkalpak — APPROVED** [review]
I can't claim to have read through each line of code, but here are a couple of things I ran into:

- There are more newlines added than necessary. I assume that is OK.
- Many code blocks have trailing empty lines (which looks really bad when line numbers are shown too).
- Clicking on "Resources" in the sidenav both navigates to (the non-existent) `/resources` and expands the "Resources" submenu. This is different thn what we have in staging right now, where the submenu has been renamed to "References" and there is link to resources in the top-menu.
- The issue with floating images isn't fixed (but I assume it was outside the scope of this PR).

I am not sure if any of these issues are even related to this PR, so I am happy to tackle them in follow-up PRs.

File: aio/content/examples/.gitignore [hunk 3]
```diff
@@ -23,12 +23,18 @@ protractor-helpers.js
 dist/
 
 # special
-!_boilerplate/
 !/*
```
Comment (petebacondarwin) [inline aio/content/examples/.gitignore]: We don't want to ignore files that are directly inside the examples folder.

File: aio/content/examples/.gitignore [hunk 4]
```diff
@@ -23,12 +23,18 @@ protractor-helpers.js
...
-!systemjs.config.*.js
 !*.1.*
 !*.2.*
 !*.3.*
```
Comment (petebacondarwin) [inline aio/content/examples/.gitignore]: There are a bunch of file in the examples that are custom versions of the boilerplate.
For example `system.config.js` is boilerplate and should be ignored. But `system.config.1.js`, etc is a custom file that is part of the example and should not be ignored. The idea of using a number to indicate a custom file is conventional in these examples so this captures up to three of them.

**Comment by petebacondarwin** [comment]
> There are more newlines added than necessary. I assume that is OK.

You can never have too many newlines (at least outside of preformatted HTML). It was easier to add in extra unnecessary newlines, which mostly are ignored, than to somehow deduce the exact combination of lines of markdown that is optimal for the myriad of variations in the docs.

> Many code blocks have trailing empty lines (which looks really bad when line numbers are shown too).

That is not good, but I think will have to be left to manual intervention once we switch over from automated migration, unless you can spot a pattern that we can include in the migrator.

> Clicking on "Resources" in the sidenav both navigates to (the non-existent) /resources and expands the "Resources" submenu. This is different thn what we have in staging right now, where the submenu has been renamed to "References" and there is link to resources in the top-menu.

I deleted the resources file under agreement from @wardbell that he is creating a new one manually in the marketing pages. So hopefully this will fix itself very soon.

> The issue with floating images isn't fixed (but I assume it was outside the scope of this PR).
That is a styling issue, which is outside the scope of this PR.

File: aio/content/glossary.md (hunk 1 above)
Comment (petebacondarwin) [inline aio/content/glossary.md]: You are right. Unfortunately this is a case of Jade that we cannot cope with. There is a div that is marked as a sub-section containing a div that is marked as an alert.

**Simplified example:**

```
~~~ {.sub-section}
  some markdown formatted text
  ~~~ {.alert}
    some more markdown
  ~~~
~~~
```

While we have the ability to create "components", which are styled divs containing markdown, we cannot support nested "components".

Comment (petebacondarwin) [inline aio/content/glossary.md]: @wardbell and I agreed that this was a fairly rare case that would need to be manually dealt with.

File: aio/content/examples/.gitignore (hunk 3 above)
Comment (gkalpak) [inline aio/content/examples/.gitignore]: Yes, but what would be causing these files to be ignored?

Comment (gkalpak) [inline aio/content/examples/.gitignore]: Nevermind. Now that I see the whole picture it makes sense :grin:

Comment (petebacondarwin) [inline aio/content/examples/.gitignore]: All the lines from 2-13 ignore boilerplate stuff that we then need to unignore.

**Pull request merged**